src/utils/avdlToAVSC.ts

```diff
@@ -1,12 +1,31 @@
 import * as fs from 'fs'
-import { AssembleProtocolError, assembleProtocol, readProtocol } from 'avsc'
+import { assembleProtocol, readProtocol } from 'avsc'
+
 import { ConfluentSchemaRegistryError } from '../errors'
 
+interface AssembleProtocolError extends Error {
+  path: string
+}
+interface Obj {
+  [key: string]: any
+}
+interface Iterable extends Obj {
+  map: any
+}
+interface Field {
+  type: {
+    type: string
+    items: any
+  }
+}
+
 let cache: any
 const merge = Object.assign
-const isObject = (obj: any) => obj && typeof obj === 'object'
-const isIterable = (obj: any) => isObject(obj) && typeof obj.map !== 'undefined'
-const isFieldArray = (field: any) => isObject(field.type) && field.type.type === 'array'
+const isObject = (obj: unknown): obj is Obj => obj && typeof obj === 'object'
+const isIterable = (obj: unknown): obj is Iterable =>
+  isObject(obj) && typeof obj.map !== 'undefined'
+const isFieldArray = (field: unknown): field is Field =>
+  isObject(field) && isObject(field.type) && field.type.type === 'array'
```

Comment:
Except for the typings,`isObject(field)` is the only _actual_ code change. Am I correct in assuming that `field` is an object? 🙂

Reply:
That is correct